src/blockdev.rs

```diff
@@ -566,39 +597,58 @@ impl SavedPartitions {
             });
         }
 
+        // Create a vector of the partitions
+        let mut partitions = Vec::new();
+
         // read GPT
-        let gpt = match GPT::find_from(disk) {
-            Ok(gpt) => gpt,
+        match GPT::find_from(disk) {
+            Ok(gpt) => {
+                // cross-check GPT sector size
+                Self::verify_gpt_sector_size(&gpt, sector_size)?;
+
+                // save partitions accepted by filters
+                for (i, p) in gpt.iter() {
+                    if Self::matches_filters(i, p, filters) {
+                        partitions.push((i, p.clone()));
+                    }
+                }
+            }
             Err(gptman::Error::InvalidSignature) => {
-                // ensure no indexes are listed to be saved from a MBR disk
-                // we don't need to check for labels since MBR does not support them
+                // no GPT, check for MBR
                 if filters
                     .iter()
                     .any(|f| matches!(f, PartitionFilter::Index(_, _)))
                     && disk_has_mbr(disk).context("checking if disk has an MBR")?
                 {
-                    bail!("saving partitions from an MBR disk is not yet supported");
+                    let mbr = MBR::read_from(disk, u32::try_from(sector_size)?)
+                        .context("reading disk as MBR")?;
+
+                    // cross-check MBR sector size
+                    Self::verify_mbr_sector_size(&mbr, sector_size)?;
+
+                    for (i, p) in mbr.iter() {
+                        if Self::matches_filters_mbr(i.try_into().unwrap(), p, filters) {
+                            // if the partition is using any of the last 33 sectors, we need to bail.
+                            if p.starting_lba + p.sectors < mbr.disk_size - 33 {
```

Comment:
May be worth hosting `33` into a `const FINAL_SECTORS: u32 = 33` above?

Reply:
We'd also need a comment explaining how we arrived at 33 sectors.  But note also that MBR is supported on 4Kn-sector disks (they just aren't bootable).  The backup GPT requires fewer sectors in that case.

GPTs are allowed to reserve more than the minimum number of sectors at either end of the disk.  We don't really check for that in the non-MBR case either, and don't have a way to, since we haven't seen the install image's partition table.  But I believe in that case we'd fail to restore partitions after installing and would then roll back to a fresh GPT with only the saved partitions, thus not losing data.

Long story short, I think we should probably skip this check.  The dry run below should catch overlarge partitions if there are any.  (But we should still have a test case for this.)